frappe/integrations/doctype/integration_request/integration_request.py

```diff
@@ -20,3 +22,20 @@ def update_status(self, params, status):
 		self.status = status
 		self.save(ignore_permissions=True)
 		frappe.db.commit()
+
+	def handle_sucess(self, response):
+		"""update the output field with the response along with the relevant status"""
+		self.process_response("output", response)
+
+	def handle_failure(self, response):
+		"""update the error field with the response along with the relevant status"""
+		self.process_response("error", response)
+
+	def process_response(self, ref_field, response):
```

Comment:
Maybe I was not clear as to what I meant. `process_response` does not tell me what it is going to **process** but `handle_success` tells me that it will handle a success response. `process_response` should not exist in this class as it is trying to do multiple things and is ambigous. 

Check [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single-responsibility_principle)

Reply:
added